### PR TITLE
Add Documentation for Docker compose Manual Installation without using the `install.sh` script

### LINKED ISSUE #2761: Provide documentation for running Mathesar without the bash script
## Problem
Currently, installation of Mathesar requires running of a bash script from the internet which is highly insecure and requires in-depth bash knowledge to inspect the script before running it.

## Proposed solution
Provide documentation to run Mathesar via docker or via docker-compose.yml in conjunction with environmental variables.

## Additional context
See my question here: https://www.reddit.com/r/selfhosted/comments/11n2fxx/comment/jbomlgl/ 
See me being asked to open an issue here: 
https://github.com/centerofci/mathesar/issues/2034#issuecomment-1486932641 

## Suggested solution: 
https://github.com/centerofci/mathesar/issues/2741#issuecomment-1482598608

### PR DESCRIPTION
**Pull request by silentninja**
Fixes #2761 

Adds Manual Installation Section to the docker-compose installation documentation.

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `develop` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by silentninja** [comment]
@mathemancer or @seancolsen Can anyone of you review the PR in addition to the review done by @pavish

**Comment by seancolsen** [comment]
@silentninja This PR seems to be adding a new page at `docs/docs/install/docker/index.md`, but I don't see it in the `mkdoc.yml` nav menu. Can you add it?

**Comment by silentninja** [comment]
> Is this PR adding a new page at `docs/docs/install/docker/index.md`? If so, I don't see it in the `mkdoc.yml` nav menu. Can fix this?

My bad, the docker installation branch accidentally crept into this branch. I removed those commits, thanks for the catch.

File: docs/docs/install/docker-compose/index.md [hunk 1]
```diff
@@ -44,7 +48,145 @@
...
+## Manual Install
+
+1. Open the Mathesar configuration directory where you'd like to install Mathesar. By convention, we do it within `/etc/mathesar`
+
+2. Download the necessary files into the configuration directory
+
+    - Download the `docker-compose.yml` file from [https://github.com/centerofci/mathesar/raw/master/docker-compose.yml](https://github.com/centerofci/mathesar/raw/master/docker-compose.yml)
+     - Download the example `.env.example` file used for setting the [configuration variables](../configuration.md) from [https://github.com/centerofci/mathesar/raw/master/.env.example](https://github.com/centerofci/mathesar/raw/master/.env.example)
+     - Rename the downloaded `.env.example` to `.env`
+     - Download the `Caddyfile` from [https://github.com/centerofci/mathesar/raw/master/Caddyfile](https://github.com/centerofci/mathesar/raw/master/Caddyfile).
```
Comment (silentninja) [inline docs/docs/install/docker-compose/index.md:61]: This step is not needed as our prebuilt docker image already has the Caddyfile

**Review by seancolsen — APPROVED** [review]
@silentninja I added quite a bit of clean up in d2d23e441adf85604ace7d76ce01839301d5ddb7. I'm happy to address any questions or concerns with my changes.

As [discussed on Matrix](https://matrix.to/#/!UZILDSNKobkelUYwBp:matrix.mathesar.org/$0UCBIcMdyWCWa6zB_t7OnUyR5oJUok4Pm2aqCf2Kem0?via=matrix.mathesar.org&via=matrix.org), I think it would be good to help the reader understand what OS user should own the `/etc/mathesar` directory. We can make that improvement in a subsequent PR though.

File: docs/docs/install/docker-compose/index.md (hunk 1 above)
Comment (seancolsen) [reply docs/docs/install/docker-compose/index.md]: I addressed this in d2d23e441adf85604ace7d76ce01839301d5ddb7

Comment (silentninja) [reply docs/docs/install/docker-compose/index.md]: Thanks!

**Review by pavish — APPROVED** [review]
@silentninja Looks good to me!

File: docs/docs/install/docker-compose/index.md [hunk 2]
```diff
@@ -49,37 +49,56 @@
...
+## Manual Install {#manual}
+
+1. Navigate to a directory where you'd like to store your Mathesar configuration. By convention, we do it within `/etc/mathesar`, but it can be a different directory if you like.
+
+    ```
+    sudo mkdir -p /etc/mathesar
+    cd /etc/mathesar
+    ```
+
+1. Download our [docker-compose.yml](https://github.com/centerofci/mathesar/raw/master/docker-compose.yml), and [.env.example](https://github.com/centerofci/mathesar/raw/master/.env.example) files to your configuration directory.
```
Comment (silentninja) [inline docs/docs/install/docker-compose/index.md:61]: @seancolsen Is there a reason why the items are not numbered correctly and instead are using `1` for every item?

**Pull request merged**